ddl/partition.go

```diff
@@ -890,6 +905,13 @@ func onDropTablePartition(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		}
 	}
 
+	rules := buildPlacementDropRules(job.SchemaID, tblInfo.ID, physicalTableIDs)
+
+	err = infosync.UpdatePlacementRules(nil, rules)
+	if err != nil {
+		return ver, errors.Wrapf(err, "failed to notify PD the placement rules")
+	}
+
```

Comment:
Is it able to rollback?

Reply:
PD will rollback in case of an error. So a failure means nothing happened on PD. While it is not 100% safe(https://github.com/xhebox/pd/blob/delete/server/schedule/placement/rule_manager.go#L294-L299), it is safe enough for now.

Reply:
What if `updateVersionAndTableInfo` succeeds but `UpdatePlacementRules` failed?

Reply:
That sounds like a trouble... It is fixed.

Reply:
What if `UpdatePlacementRules` succeeds but `updateVersionAndTableInfo` fails?

Reply:
Oh, the job is not rollbackable, so it will retry infinitely.

Reply:
What is your idea? What about a middle state? So it will only retry the `updateVersionAndTableInfo` function.